quad/src/semi_infinite_function.rs

```diff
@@ -1,58 +1,34 @@
 use crate::constants::*;
+use ndarray::Array1;
 
-pub fn semi_infinite_function<F>(f: &F, x: f64, start: f64, infty: f64) -> Vec<f64>
+pub fn semi_infinite_function<F>(f: &F, x: f64, start: f64, infty: f64) -> Array1<f64>
 where
-    F: Fn(f64) -> Vec<f64> + ?Sized,
+    F: Fn(f64) -> Array1<f64> + ?Sized,
 {
     if x < UFLOW.sqrt() {
-        return vec![0.0; f(0.0).len()];
+        return Array1::<f64>::from_vec(vec![0.0; f(0.0).len()]);
```

Comment:
Try with [`Array::zeros`](https://docs.rs/ndarray/0.15.6/ndarray/struct.ArrayBase.html#method.zeros), in this way you would avoid the creation of an intermediate vector.